test/dynamo/test_aot_autograd.py

```diff
@@ -489,6 +638,71 @@ def guard_fail_fn(failure):
         self.assertEqual(cc.frame_count, 2)
         self.assertEqual(failure_reason, "c is d")
 
+    @patch("torch._functorch.config.debug_assert", True)
+    def test_multiple_aot_autograd_calls_dupe_args(self):
+        def maybe_dupe_op(x):
+            y = x + 1
+            z = x + 2
+            if x.numel() < 5:
+                return y, y
+            else:
+                return y, z
+
+        aten = torch.ops.aten
+        lib = torch.library.Library("custom", "DEF")
+        lib.define("maybe_dupe_op(Tensor a) -> (Tensor, Tensor)")
+        lib.impl("maybe_dupe_op", maybe_dupe_op, "CPU")
+        lib.impl("maybe_dupe_op", maybe_dupe_op, "Meta")
+
+        # this is just dealing with the fact that
+        # aot_module_simplified expects submods to always return tuples/lists
+        class WrapperModule(torch.nn.Module):
+            def __init__(self, mod):
+                super().__init__()
+                self.mod = mod
+
+            def forward(self, *args):
+                out = self.mod(*args)
+                if isinstance(out, (list, tuple)):
+                    return out
+                return (out,)
+
+        def compile_submod(input_mod, args):
+            from functorch.compile import nop
+            from torch._functorch.aot_autograd import aot_module_simplified
+
+            class WrapperModule(torch.nn.Module):
+                def __init__(self):
+                    super().__init__()
+                    self.original = input_mod
+                    self.submod = aot_module_simplified(input_mod, args, nop)
+
+                def forward(self, *args):
+                    return self.submod(*args)
+
+            return WrapperModule()
+
+        def test_compile(fx_g, example_inps):
+            split_gm = torch.fx.passes.split_module.split_module(
+                fx_g, None, lambda node: 1 if "mul" in str(node) else 0
+            )
+            submod_1_inps = split_gm.submod_0(*example_inps)
+            split_gm.submod_0 = compile_submod(
+                WrapperModule(split_gm.submod_0), example_inps
+            )
+            split_gm.submod_1 = compile_submod(
+                WrapperModule(split_gm.submod_1), submod_1_inps
+            )
+            return split_gm
+
+        @torch._dynamo.optimize(test_compile)
+        def f(a):
+            b, c = torch.ops.custom.maybe_dupe_op(a)
+            return (b.mul_(c),)
+
+        f(torch.ones(4))
```

Comment:
i guess in this case just 'not throwing an exception' was enough, but should we check the output just in case?

Reply:
Yea, other tests would catch correctness I think, but we can add that here, won't hurt.